### ISSUE TITLE
when using devenv with a flake "devunv up" does not work

### ISSUE THREAD

**Issue by PeterZainzinger** [cachix/devenv#229]
**Describe the bug**

As the title suggests process dont work (for me at least) when using `devenv` from a flake.

**To Reproduce**

* create a flake with `nix flake init --template github:cachix/devenv`
* add some process, eg:
```
processes = {
    silly-example.exec = "while true; do echo hello && sleep 1; done";
    ping.exec = "ping example.com";
  };
```
* run `devenv up`

output: 

```
... error: path '/home/peter/new_stack/devenv_try/devenv.nix' does not exist in Git repository 'REDACTED_PATH_TO_GIT_REPO'
```

It seems that the flake use case is not fully supported yet.

**Version**
 ```
devenv --version                                                                                                                          
https://devenv.sh (version 0.4): Fast, Declarative, Reproducible, and Composable Developer Environments
```

**Comment by domenkozar** [comment]
cc @thenonameguy @bobvanderlinden, maybe we could expose some commands like devenv-up inside the environment?

**Comment by bobvanderlinden** [comment]
Not sure how the subcommands are going to be implemented. But if those work in the same manner as git, then `devenv-up` makes sense. It can be used from `devenv up` as well as `devenv-up` directly.

Having devenv always make `devenv-up` be part of the PATH and not having it be part of the devenv executable itself would make this more generic. Also opens up the possibility to extend devenv further.

**Comment by domenkozar** [comment]
Possible solution https://github.com/cachix/devenv/issues/233

**Comment by thenonameguy** [comment]
@PeterZainzinger for now this is how I solved the CLI usage problem:
https://github.com/snamiki1212/realworld-v1-rust-actix-web-diesel/blob/773d7711f37e988b01935c622683533cd9178f2a/flake.nix#L35
Instead of specifying everything in the flake, you still need to specify the `devenv` module in the 'flake-less' filestructure.
I'll take a look at having a more self-contained devenv cli version, while using it from a Flake over the holidays.

**Comment by PeterZainzinger** [comment]
@thenonameguy 

I came up with the same approach myself. It works ok for now.  What I am unsure is what version of nixpkgs is than actually used. Lets say a specific commit is selected as an input of the flake, what is than actually used by devenv? I would expect that it is the exact same version as specified in the flake, but I suspect it is not the case.

**Comment by thenonameguy** [comment]
Yup, it's a bit of a mess, having to manage 2 flake files within one logical git repo. I will try to do a rewrite of the CLI that works with a single one, instead of the status quo.

**Comment by domenkozar** [comment]
This is fixed in master, flake integration now provides a shim `devenv` with `up` command

**Comment by bobvanderlinden** [comment]
Oo awesome! Nice solution 👍
